### ISSUE TITLE
Player.cpp is too big, it should be split

### ISSUE THREAD

**Issue by FrancescoBorzi** [azerothcore/azerothcore-wotlk#6698]
This is causing troubles with several editors and CPU going very high when working on this file. It's definitely not a good practice to have such a huge file.

I still haven't looked at how to split it so suggestions are welcome.

**Comment by iraizo** [comment]
I tested editing both files with micro,nano,visual studio code and i have no problems but nearly 30000 lines should be split, my suggestion would be first of all splitting all setters and getters into 2 files, `player_setters` and `player_getters` (i havent looked at the naming convetion yet so). Another one could be having player_checks where the `has...` functions could be put in like `Player::HasTankSpec`. The same thing could be done with the update timers, same goes with the `Load...` functions and `Send...` functions.

**Comment by Bogir** [comment]
PlayerDefines.cpp

**Comment by iraizo** [comment]
> PlayerDefines.cpp

I mean.. the whole file is just function defenitions

**Comment by buddiman** [comment]
Maybe categorize the functions, when there are more than let's say 20 functions for one category create a new file.
For Example: 

- Quest related functions into "PlayerQuest.cpp" 
- Equipment related functions into "PlayerEquipment.cpp" 
- Spell / Talent related function into "PlayerSpell.cpp"

This would also help to follow Clean Code principles and results in more readable and mostly, more maintainable code.

**Comment by FrancescoBorzi** [comment]
So I've started this process by first extracting the existing categories of methods:

https://github.com/azerothcore/azerothcore-wotlk/pull/6838

Of course, this is not yet ideal and there are still a lot of methods out of place, but it's a start and makes it already way less messy.

If you want to move more methods into those existing categories or create new categories, PRs are welcome.

**Comment by iraizo** [comment]
> 
> 
> So I've started this process by first extracting the existing categories of methods:
> 
> #6838
> 
> Of course, this is not yet ideal and there are still a lot of methods out of place, but it's a start and makes it already way less messy.
> 
> If you want to move more methods into those existing categories or create new categories, PRs are welcome.

should the PR be made on the fork or this repo?

**Comment by FrancescoBorzi** [comment]
> > So I've started this process by first extracting the existing categories of methods:
> > #6838
> > Of course, this is not yet ideal and there are still a lot of methods out of place, but it's a start and makes it already way less messy.
> > If you want to move more methods into those existing categories or create new categories, PRs are welcome.
> 
> should the PR be made on the fork or this repo?

@iraizo no, you can target the main AC repo. But it's better if you merge my branch into yours, so your changes will not conflict with mine.

Thanks for your collaboration!

**Comment by iraizo** [comment]
> > > So I've started this process by first extracting the existing categories of methods:
> > > #6838
> > > Of course, this is not yet ideal and there are still a lot of methods out of place, but it's a start and makes it already way less messy.
> > > If you want to move more methods into those existing categories or create new categories, PRs are welcome.
> > 
> > 
> > should the PR be made on the fork or this repo?
> 
> @iraizo no, you can target the main AC repo. But it's better if you merge my branch into yours, so your changes will not conflict with mine.
> 
> Thanks for your collaboration!

I will fix the merge conflicts myself if any exist after yours or mine will be merged since i pulled your fork, then checked it out on my fork but i probaly fucked it up, im going to continue the clean up process.

**Comment by iraizo** [comment]
Isn't this already done now? @FrancescoBorzi

**Comment by FrancescoBorzi** [comment]
@iraizo we improved it a little bit indeed but I think it's still too big

**Comment by iraizo** [comment]
> @iraizo we improved it a little bit indeed but I think it's still too big

👍🏻
